blocks/tabs/tabs.js

```diff
@@ -0,0 +1,150 @@
+import { createOptimizedPicture } from '../../scripts/aem.js';
+import {
+  domEl,
+  div,
+  ul,
+  li,
+  button,
+} from '../../scripts/dom-helpers.js';
+
+export default function decorate(block) {
+  document.querySelector('.tabs-container').classList.add('calcite-mode-dark');
+
+  block.querySelectorAll('img').forEach((img) => img
+    .closest('picture')
+    .replaceWith(
+      createOptimizedPicture(img.src, img.alt, false, [{ width: '750' }]),
+    ));
+
+  const tabTitles = [...block.children].map((child) => child.children[0].children[0].textContent);
+  const tabContents = [...block.children].map((child) => [...child.children[1].children]);
+
+  tabContents.forEach((content) => {
+    const text = [content[1], content[2], content[3]];
+    const textWrapper = div({ class: 'text-wrapper' }, ...text);
+    content.splice(1, 3, textWrapper);
+
+    const hrefs = [content[2].children[0].href, content[3].children[0].href];
+    const buttons = [
+      domEl('calcite-button', {
+        'icon-end': 'play-f',
+        href: hrefs[0],
+        appearance: 'solid',
+        alignment: 'center',
+        scale: 'm',
+        type: 'button',
+        width: 'auto',
+        kind: 'inverse',
+        color: 'inverse',
+        'calcite-hydrated': '',
+      }, content[2].textContent),
+      domEl('calcite-button', {
+        'icon-end': 'arrowRight',
+        href: hrefs[1],
+        appearance: 'outline',
+        alignment: 'center',
+        scale: 'm',
+        type: 'button',
+        width: 'auto',
+        kind: 'inverse',
+        'calcite-hydrated': '',
+      }, content[3].textContent),
+    ];
+
+    const buttonsWrapper = div({ class: 'buttons-wrapper' }, ...buttons);
+    content.splice(2, 2, buttonsWrapper);
+  });
+
+  const contents = tabContents.map((content) => div({ class: 'tab-content' }, ...content));
+  const titles = tabTitles.map((title) => li({
+    class: 'tab-title',
+    id: title.toLowerCase().replace(' ', '-'),
+    'aria-hidden': true,
+    'aria-selected': false,
+  }, button(...title)));
+
+  const arrowLeft = domEl(
+    'calcite-button',
+    {
+      class: 'arrow-button left',
+      'icon-end': 'chevronLeft',
+      scale: 'l',
+      kind: 'neutral',
+      round: '',
+    },
+  );
+
+  const arrowRight = domEl(
+    'calcite-button',
```

Comment:
given how much calcite-button is used you might want to create a dom-helper for it